sycl/test-e2e/ESIMD/unified_memory_api/Inputs/scatter.hpp

```diff
@@ -267,14 +267,11 @@ bool testSLM(queue Q, uint32_t MaskStride,
          constexpr uint32_t SLMSize = (Threads * N + 8) * sizeof(T);
          slm_init<SLMSize>();
 
-         if (LocalID == 0) {
-           for (int I = 0; I < Threads * N; I += 8) {
-             simd<T, 8> InVec(Out + GlobalElemOffset + I);
-             simd<uint32_t, 8> Offsets(I * sizeof(T), sizeof(T));
-             slm_scatter<T>(Offsets, InVec);
-           }
+         for (int I = 0; I < Threads * N; I += 8) {
+           simd<T, 8> InVec(Out + GlobalElemOffset + I);
+           simd<uint32_t, 8> Offsets(I * sizeof(T), sizeof(T));
+           slm_scatter<T>(Offsets, InVec);
          }
-         barrier();
 
```

Comment:
The code "before" looks correct, and the new code is potentially UB, because many threads write to the same memory.

Reply:
Ok, I see now that Threads = 1, which makes this code correct. But it looks very confusing.
If look only to this code (and forget about Threads being set to 1), then this code looks like UB

Reply:
If there is some error in GPU Driver, then let's have Threads = 1 for now and keep this code, but please add a TODO comment here explaining that if Thread > 1, then this code needs an initialization done by only 1 thread and need a barrier between SLM initialization and the usage of SLM by all threads of the work-group.